remote/remote.go

```diff
@@ -93,6 +102,18 @@ func AddEmptyLayerOnSave() ImageOption {
 	}
 }
 
+// WithRegistry register a registry.
+func WithRegistry(repository string, insecure, insecureSkipVerify bool) ImageOption {
```

Comment:
Would it be possible to add a test that calls `NewImage()` with this option?

Reply:
I have no good idea to add a test for this option.
If it wants to verify the insecure option， it needs a docker registry with a non-localhost IP  or a non-private internets  IP, or a domain, this is not easy to achieve under the existing testing framework.

Do you have any suggestions?

Reply:
Ah, you are right. Sigh... I suppose we could make the field on the Image struct public, but that feels unsatisfying. I'd be okay with not adding a test in this case.